src/erlcloud_ec2.erl

```diff
@@ -1441,27 +1538,131 @@ extract_block_device_mapping_status(Node) ->
      {delete_on_termination, get_bool("ebs/deleteOnTermination", Node)}
     ].
 
-%%
-%%
--spec(describe_instance_status/1 :: (string()) -> proplist()).
-describe_instance_status(InstanceID) ->
-    describe_instance_status([{"InstanceId", InstanceID}], [], default_config()).
+-type instance_id() :: string().
+-type ec2_param_list() :: [{string(),string()}].
 
--spec(describe_instance_status/2 :: (proplist(), filter_list()) -> proplist()).
-describe_instance_status(Params, Filter) ->
-    describe_instance_status(Params, Filter, default_config()).
+-define(SPOT_INSTANCE_STATUS_MR_MIN, 5).
+-define(SPOT_INSTANCE_STATUS_MR_MAX, 1000).
 
--spec(describe_instance_status/3 :: (proplist(), filter_list(), aws_config()) -> proplist()).
-describe_instance_status(Params, Filter, Config) ->
+-spec describe_instance_status() -> ec2_normal_describe_return().
+describe_instance_status() ->
+    describe_instance_status([], [], default_config()).
+
+-spec describe_instance_status(instance_id()) -> ec2_normal_describe_return();
+                              (aws_config()) -> ec2_normal_describe_return();
+                              (ec2_max_result()) -> ec2_paged_describe_return().
+describe_instance_status(InstanceID)
+    when is_list(InstanceID) ->
+    describe_instance_status([{"InstanceId", InstanceID}], [], default_config());
+describe_instance_status(Config)
+    when is_record(Config, aws_config) ->
+    describe_instance_status([], [], Config);
+describe_instance_status(MaxResults)
+    when is_integer(MaxResults) ->
+    describe_instance_status([], [], MaxResults, undefined, default_config()).
+
```

Comment:
agree to the first part and disagree to the last part with more tuple wrapping. what for? and again rest of the modules have it with tuple of three elements so let's not produce more snowflakes.